sharktank/tests/models/llama4/llama4_test_iree_eager.py

```diff
@@ -0,0 +1,411 @@
+        str(batch_size),
+        "--block-seq-stride",
+        str(config.block_seq_stride),
+        "--attention-dtype",
+        "float16",
+        "--activation-dtype",
+        "float16",
+        "--tensor-parallelism-size",
+        "1",
+        "--pipeline-parallelism-size",
+        "1",
+        "--attention-kernel",
+        "torch",
+        "--skip-decode",
+        "--use-attention-mask",
+    ]
+
+    sys_argv_backup = sys.argv
+    sys.argv = cli_args
+    export_paged_llm_v1.main()
+
+    sys.argv = sys_argv_backup
+
+    iree_module_path = output_path / "model.vmfb"
+    export_artifacts.compile_to_vmfb(
+        output_mlir=mlir_path,
+        output_vmfb=iree_module_path,
+        args=[],
+    )
+
+
+def export_llama4_iree_parameters(
+    model: PagedLlmModelV1,
+    parameters_output_path: PathLike,
+    config: LlamaModelConfig,  # TODO; it can be read from HF when comparing the actual model
+):
+    config_dict = {
+        "hparams": asdict(config.hp),
+    }
+
+    dataset = Dataset(config_dict, root_theta=model.theta)
+    dataset.save(parameters_output_path)
+
+
+def export_llama4(
+    model: PagedLlmModelV1,
+    mlir_output_path: PathLike,
+    parameters_output_path: PathLike,
+    output_path: PathLike,
+    batch_size: int,
+    config: LlamaModelConfig,  # TODO; it can be read from HF when comparing the actual model
+):
+    export_llama4_iree_parameters(model, parameters_output_path, config=config)
+    export_llama4_model_mlir(
+        model, output_path=output_path, config=config, batch_size=batch_size
+    )
+
+
+def runCompareIreeAgainstTorchEager(args, atol):
+    seed = 1234
+    random.seed(seed)
+    torch.manual_seed(seed)
+
+    work_dir = Path(args.work_dir)
+
+    MoeBlock.forward = patch_forward_moe
+
+    # add sample_inputs
+    PagedLlmModelV1.sample_inputs = _sample_inputs
+    PagedLlmModelV1.input_mask = staticmethod(deterministic_input_mask)
+    PagedLlmModelV1.attention_mask = staticmethod(deterministic_attn_mask)
+
+    target_theta, target_config = generate(seed)
+    target_torch_model = PagedLlmModelV1(
+        theta=target_theta,
+        config=target_config,
+    )
+
+    _, kw = target_torch_model.sample_inputs(args.batch_size)
+    logger.info("running eager prefill for llama4...")
+    expected_outputs = target_torch_model.prefill(
+        kw["tokens"],
+        attention_mask=kw["attention_mask"],
+        seq_block_ids=kw["seq_block_ids"],
+        cache_state=kw["cache_state"],
+    )
+
+    # Iree model
+    mlir_path = work_dir / "model.mlir"
+    parameters_path = work_dir / "parameters.irpa"
+
+    logger.info("Exporting llama4 to MLIR...")
+    export_llama4(
+        target_torch_model,
+        output_path=work_dir,
+        mlir_output_path=mlir_path,
+        parameters_output_path=parameters_path,
+        batch_size=args.batch_size,
+        config=target_config,
+    )
+
+    iree_devices = get_iree_devices(
+        device="hip://5",
```

Comment:
You need to use hip://0 since this will likely run on a single GPU runner.

Reply:
This test should use the `get_iree_flags` pytest fixture to acquire the target device and related compilation flags.
Then the test becomes retargetable and can run on CPU, HIP, etc.